### ISSUE TITLE
Create a new GitHub action that adds/removes "community-contribution-in-progress" label on issues

### ISSUE THREAD

**Issue by MisRob** [learningequality/kolibri#12923]
## Summary

**Create a new GitHub action that adds/removes _"community-contribution-in-progress"_ label on issues.**

![Image](https://github.com/user-attachments/assets/ef27e307-9e3c-454f-9436-3991804a3939)

- When an external contributor is assigned to an issue, the action should add the _"community-contribution-in-progress"_  label to the issue
- When an external contributor is unassigned from an issue, the action should remove the _"community-contribution-in-progress"_  label

**This needs to be a shared action with most of its logic placed in [the .github repository](https://github.com/learningequality/.github).** This shared part will then be called from the following repositories:

A. [kolibri](https://github.com/learningequality/kolibri)
B. [kolibri-design-system](https://github.com/learningequality/kolibri-design-system)
C. [studio](https://github.com/learningequality/studio)
D. [ricecooker](https://github.com/learningequality/ricecooker/)
E. [le-utils](https://github.com/learningequality/le-utils/)

### Development process

- For ease of development, at first work on your own two repositories
- Test as many areas as possible manually

Later during the review process, we can offer guidance in areas specific to integration with our ecosystem, such as private secrets, etc.

### Opening pull requests

To allow for smooth review and testing process, we'd ask you to organize pull requests as follows:

- (1) The 1st pull request with the re-usable shared logic to target [the `.github` repository](https://github.com/learningequality/.github)
- (2) The 2nd pull request with a small action that re-uses (1) to target [the `test-actions` repository](https://github.com/learningequality/test-actions). Imagine `test-actions` as a temporary representative of (A-E) repositories listed above. 

Only after we have reviewed the two pull requests above, the final 5 pull requests can be open to (A-E) repositories. You will be instructed by a team member when the time comes to do so.

See the following action for an example:

- [`test-actions` repository part](https://github.com/learningequality/test-actions/blob/main/.github/workflows/notify_team_new_comment.yml)
- [`.github` repository part](https://github.com/learningequality/.github/blob/main/.github/workflows/notify_team_new_comment.yml)

### 3rd party workflows

- For security and upgrade reasons, we prefer our own implementation together with official 3rd party workflows if needed.

### Programming language

- Generally for more complex actions, use JavaScript or Python scripts rather than bash.

### Implementation

- To identify external contributors, use their association: it shouldn't be `MEMBER` and neither `OWNER`

## Acceptance criteria

- [ ] The action adds/removes the label when an external contributor is assigned/unassigned an issue
- [ ] The action ignores issues assigned/unassigned to core team members

**Comment by adibmbrk** [comment]
Hi @MisRob, can i get this issue assigned to me?

**Comment by iamshobhraj** [comment]
hi @MisRob,  can i get this assigned?

**Comment by MisRob** [comment]
Hi @iamshobhraj, I will assign you, thank you.

@adibmbrk Thanks for volunteering! I'm sorry, I offered the issue on some of our channels where @iamshobhraj is a member due to some previous contribution, so will reserve this. I believe there may still be other 'help wanted' and we will add more next year, if that'd be of your interest.

**Comment by MisRob** [comment]
Hi @iamshobhraj,  I wanted to mention that [Learning Equality will be closed from December 23 to January 5](https://github.com/learningequality/kolibri/discussions/12956).

**Comment by iamshobhraj** [comment]
Happy New Years, @MisRob ! 🎉

I've completed the workflow that manages labels on issues with external contributors. It’s working perfectly in my personal repository, [test-actions](https://github.com/iamshobhraj/test-actions). This workflow leverages reusable logic from another repository of mine, [.github](https://github.com/iamshobhraj/.github).

Since I had to use this GitHub Action in my personal repository, I created and used GitHub tokens. For the organization, we'll need to ensure we have a similar token set up if it isn’t already covered.

**Comment by marcellamaki** [comment]
Hi @iamshobhraj - thank you for the update and your contribution! We do have github tokens that we use for many of our github actions, but I am not sure about the specifics of this use case, and what might be required for properly configuring that in your PR. @MisRob will be out of the office for the new year still until next week, and she will follow up with you after her return. Happy new year!

**Comment by MisRob** [comment]
Hi @iamshobhraj, that's wonderful, thanks. You're welcome to open pull requests to those two repositories and we will coordinate with you on tokens.

**Comment by iamshobhraj** [comment]
@MisRob i have opened pull requests in those two repositories.

**Comment by MisRob** [comment]
Thanks @iamshobhraj!

**Comment by MisRob** [comment]
@iamshobhraj @rtibbles I have just successfully tested the action on [this issue](https://github.com/learningequality/test-actions/issues/65) and all works smoothly, so wonderful! I would feel good moving this to production and then closing this.

@iamshobhraj this is a bit cumbersome part as it requires opening several same PRs with [this change](https://github.com/learningequality/test-actions/pull/62/files) to the five repositories I mentioned in this issue's description, @iamshobhraj would you like to complete this or should I take care of it later? Let us know what works for you best. Thanks a lot for all this work.

**Comment by iamshobhraj** [comment]
@MisRob Thanks! I’m glad everything is working smoothly. I’d be happy to complete this and open the PRs for the five repositories.

**Comment by MisRob** [comment]
Thank you @iamshobhraj, feel free to go ahead any time. I left one relevant note [here](https://github.com/learningequality/test-actions/pull/62/files#r1970578505).

**Comment by iamshobhraj** [comment]
@MisRob I have opened pull requests to the five repositories mentioned in issue along with the changes you requested.

**Comment by MisRob** [comment]
Thank you @iamshobhraj!

**Comment by MisRob** [comment]
Wonderful work @iamshobhraj, thank you. We've merged all the PRs. I will now observe it in production for some time. If I encounter any issues, will link them here, but for now I think I can close.
